### ISSUE TITLE
Distributed Tracing V1 with latest DTFx not working

### ISSUE THREAD

**Issue by attilah** [microsoft/durabletask-mssql#274]
I found an issue that when I set `CorrelationSettings.Current.EnableDistributedTracing=true;` then the execution throws a NRE in `TaskOrchestrationDispatcher` [here](https://github.com/Azure/durabletask/blob/durabletask.core-v3.0.0/src/DurableTask.Core/TaskOrchestrationDispatcher.cs#L517) as `CorrelationTraceContext.Current` is `null`.

In AzureStorage I see [TraceContext](https://github.com/Azure/durabletask/blob/durabletask.core-v3.0.0/src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs#L790) set, but in the [SqlOrchestrationService](https://github.com/microsoft/durabletask-mssql/blob/main/src/DurableTask.SqlServer/SqlOrchestrationService.cs#L303) I do not.

Am I missing something and this provider requires additional setup beside the OTEL setup or the provider's distributed tracing support is not compatible with 3.0 DurableTask.Core?

**Comment by cgillum** [comment]
The `CorrelationSettings` APIs are what we consider distributed tracing v1, which was implemented based on the W3C distributed tracing spec in the Azure Storage provider but not in any of the others. We've since moved away from this and implemented distributed tracing v2, based on System.Diagnostics and OTEL. This MSSQL provider supports distributed tracing v2. The distributed tracing feature you're using now should be considered deprecated.

Distributed Tracing is essentially enabled by default when `Activity.Current` has a value and when there is a trace listener running in the process. See [here](https://github.com/microsoft/durabletask-mssql/blob/e5155826631f8abc8852c1adeb41e958ca4870d7/test/DurableTask.SqlServer.Tests/Integration/Orchestrations.cs#L696-L707) for some integration test code that shows this working.

Note that the distributed tracing v2 also works with the Azure Storage provider, and is the recommended approach for doing distributed tracing with DTFx.
